### ISSUE TITLE
Parallelize e2e tests

### ISSUE THREAD

**Issue by srdtrk** [srdtrk/cw-ica-controller#3]
## Problem Summary

Rather than having one monolithic e2e test, split it up into smaller tests that can be ran in parallel in github actions.

## Description

Currently, all the end to end tests are in one monolithic test called `TestIcaControllerContract` as subtests. This means github actions cannot run these tests in parallel, leading to long waiting times for github actions to complete.

**Comment by srdtrk** [comment]
Done
